tests/test_peak_core_and_tile.py

```diff
@@ -0,0 +1,197 @@
+
+from canal.util import IOSide
+
+from peak_core.peak_core import PeakCore
+
+from cgra import create_cgra, compress_config_data
+
+
+@pytest.fixture(scope="module")
+def dw_files():
+    filenames = ["DW_fp_add.v", "DW_fp_mult.v"]
+    dirname = "peak_core"
+    result_filenames = []
+    for name in filenames:
+        filename = os.path.join(dirname, name)
+        assert os.path.isfile(filename)
+        result_filenames.append(filename)
+    return result_filenames
+
+
+@pytest.fixture()
+def sequences():
+    """
+    * inputs - a triple (config_data, a, and b)
+      * config_data - bitstream for configuring the coreto perform a random instruction
+      * a, b - random input values for data0, data
+    * outputs - the expected output value given config_data, a, and b
+    """
+    core = PeakCore(PE_fc)
+    inputs = []
+    outputs = []
+    for _ in range(5):
+        # Choose a random operation from lassen.asm
+        op = random.choice([add, sub])
+        # Get encoded instruction (using bypass registers for now)
+        instruction = op(ra_mode=Mode_t.BYPASS, rb_mode=Mode_t.BYPASS)
+        # Convert to bitstream format
+        config_data = core.get_config_bitstream(instruction)
+        a, b = (BitVector.random(16) for _ in range(2))
+        inputs.append((config_data, a, b))
+        # generate expected output using model
+        outputs.append(core.wrapper.model(instruction, a, b)[0])
+    return inputs, outputs
+
+
+class BasicSequenceTester(SequenceTester, BasicTester):
+    """
+    Extend SequenceTester with BasicTester methods (e.g. reset, configure)
+    """
+
+    def __init__(self, circuit, sequences, clock, reset):
+        super().__init__(circuit, sequences, clock)
+        self.reset_port = reset
+
+
+def test_peak_core_sequence(sequences, dw_files):
+    """
+    Core level test
+    * configures core using instruction bitstream
+    * drives input values onto data0 and data1 ports
+    * checks alu_res output
+    """
+    def core_input_driver(tester, value):
+        config_data, a, b, = value
+        for addr, data in config_data:
+            tester.configure(addr, data)
+        tester.poke(tester._circuit.data0, a)
+        tester.poke(tester._circuit.data1, b)
+
+    def core_output_monitor(tester, value):
+        tester.expect(tester._circuit.alu_res, value)
+
+    core = PeakCore(PE_fc)
+    core.name = lambda: "PECore"
+    circuit = core.circuit()
+
+    inputs, outputs = sequences
+    tester = BasicSequenceTester(circuit, [
+        (core_input_driver, InputSequence(inputs)),
+        (core_output_monitor, OutputSequence(outputs))
+    ], circuit.clk, circuit.reset)
+    tester.reset()
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        for filename in dw_files:
+            shutil.copy(filename, tempdir)
+
+        tester.compile_and_run("verilator",
+                               directory=tempdir,
+                               magma_opts={"coreir_libs": {"float_DW"}},
```

Comment:
Can you please use CW instead? DW float IP doesn't work with PD anymore. Not going to have any difference here but just in case we need to port the tests to cad servers.

Reply:
Done (for reference the other peak_core tests still use DW, which was what I was basing off of, so something to be aware of in case that comes up later on)

Reply:
Thanks for the heads up. I will look into peak core tests in this repo. For lassen I have created an issue: https://github.com/StanfordAHA/lassen/issues/178